clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1581,6 +1581,10 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      * is invoked for the same partition more than once, the latest offset will be used on the next poll(). Note that
      * you may lose data if this API is arbitrarily used in the middle of consumption, to reset the fetch offsets
      *
+     * Note that, since messages are 0-indexed, {@link #seekToBeginning(Collection)} should be used to retrieve
+     * the message at offset 0.
+     *
+     * @param offset the last offset which was consumed. The next message received will have a higher offset number than this
```

Comment:
I'm not sure this is correct. We have many tests relying on seek(tp, 0) to make sure it starts fetching from offset 0. Ex: `BaseConsumerTest#testSimpleConsumption`. Could you find or write a simple test to confirm what you described here? Maybe there's a bug. Thanks.

Reply:
Thank you very much, you were correct. I have added a few tests in mock consumer meant to confirm the behaviour. These work on the assumption that the behaviour for the real consumer is the same as for the mock consumer (the mock consumer was just easier to write tests for)